timefold-solver/src/main/java/com/baeldung/timefoldsolver/ShiftScheduleApp.java

```diff
@@ -0,0 +1,54 @@
+package com.baeldung.timefoldsolver;
+
+import java.time.Duration;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+
+import ai.timefold.solver.core.api.solver.Solver;
+import ai.timefold.solver.core.api.solver.SolverFactory;
+import ai.timefold.solver.core.config.solver.SolverConfig;
+
+public class ShiftScheduleApp {
+
+    public static void main(String[] args) {
+        SolverFactory<ShiftSchedule> solverFactory = SolverFactory.create(new SolverConfig()
+                .withSolutionClass(ShiftSchedule.class)
+                .withEntityClasses(Shift.class)
+                .withConstraintProviderClass(ShiftScheduleConstraintProvider.class)
+                // The solver runs only for 5 seconds on this small dataset.
+                // It's recommended to run for at least 5 minutes ("5m") otherwise.
+                .withTerminationSpentLimit(Duration.ofSeconds(5)));
```

Comment:
Given a long enough time spent, is the solution deterministic?

Reply:
Yes, absolutely.
2 runs for the same input will yield the same best score at the same iteration (which is at roughly the same time)

Reproducibility is worth it's weight in gold when debugging NullPointerExceptions in a user's filter predicate :)